drivers/gpio/gpio_stm32.c

```diff
@@ -273,7 +273,7 @@ static int gpio_stm32_config(struct device *dev, int access_op,
 		return -EIO;
 	}
 
-	if ((flags & GPIO_INT) != 0) {
+	if (IS_ENABLED(CONFIG_EXTI_STM32) && (flags & GPIO_INT) != 0) {
```

Comment:
Works but I wonder why not using #ifdef here?

Reply:
I tried to follow `Within code, where possible, use the IS_ENABLED macro to convert a Kconfig symbol into a C boolean expression, and use it in a normal C conditional` from [Linux kernel coding style](https://www.kernel.org/doc/html/latest/process/coding-style.html?highlight=is_enabled#conditional-compilation)

Reply:
Well, tbh I hadn't paid attention this behavior had the same capability as `#ifdef` to exclude code block from codebase, and this is indeed much easier to read.
+1 and this practice should be extended.